### PR TITLE
Make sure to keep changed tracks up to date.

### PR DESCRIPTION
**Pull request by lazeratops**
This PR checks that a participant's media tracks are up to date in the DOM, replacing them on participant update if their ID has changed.

This is a fix for an issue reported in Slack where when switching to SFU mode, participant video goes black. I suspect this is due to tracks changing and not being updated when switching topology. I am unable to repro the issue with the change in this PR.

### CODE REVIEW

File: src/components/VideoTile.vue [hunk 1]
```diff
@@ -58,30 +58,63 @@ export default {
...
+      // only happen the first time the tile is initialized.
+      if (!existingTracks || existingTracks.length === 0) {
+        return new MediaStream([newTrack]);
+      }
+      if (existingTracks.length > 1) {
+        console.warn(`expected 1 track, found ${existingTracks.length}. Only using the first one.`);
+      }
+      const existingTrack = existingTracks[0];
+      // If existing track is different from the new track,
+      // removec the existing track and add the new one.
```
Comment (jamsea) [inline src/components/VideoTile.vue]: Small typo `removec` should be `remove` 👍

**Review by jamsea — APPROVED** [review]

File: src/components/VideoTile.vue (hunk 1 above)
Comment (lazeratops) [reply src/components/VideoTile.vue]: Typo-averse minds think alike :D

**Review by Regaddi — APPROVED** [review]
Changes look good to me 👍 

Is the updated `package-lock.json` intended?

**Comment by lazeratops** [comment]
> Changes look good to me 👍
> 
> Is the updated `package-lock.json` intended?

Thanks for taking a look.

Re package lock, yes it is; I ran `prune` and `i` and this is the diff I was left with, so I think it needs an update.

**Pull request merged**